sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/Aggregation.java

```diff
@@ -24,43 +28,76 @@
   private static final ThrottlingLogger logger =
       new ThrottlingLogger(Logger.getLogger(Aggregation.class.getName()));
 
-  private static final Aggregation NONE = Aggregation.make("none", unused -> null);
+  private static final Aggregation NONE = new NoAggregation();
   private static final Aggregation DEFAULT =
-      Aggregation.make(
-          "default",
-          i -> {
-            switch (i.getType()) {
-              case COUNTER:
-              case UP_DOWN_COUNTER:
-              case OBSERVABLE_SUM:
-              case OBSERVABLE_UP_DOWN_SUM:
-                return AggregatorFactory.sum(AggregationTemporality.CUMULATIVE);
-              case HISTOGRAM:
-                return AggregatorFactory.histogram(
-                    ExplicitBucketHistogramUtils.DEFAULT_HISTOGRAM_BUCKET_BOUNDARIES,
-                    AggregationTemporality.CUMULATIVE);
-              case OBSERVABLE_GAUGE:
-                return AggregatorFactory.lastValue();
-            }
-            logger.log(Level.WARNING, "Unable to find default aggregation for instrument: " + i);
-            return null;
-          });
+      new Aggregation() {
+        private Aggregation resolve(InstrumentDescriptor instrument) {
+          switch (instrument.getType()) {
+            case COUNTER:
+            case UP_DOWN_COUNTER:
+            case OBSERVABLE_SUM:
+            case OBSERVABLE_UP_DOWN_SUM:
+              return SUM;
+            case HISTOGRAM:
+              return EXPLICIT_BUCKET_HISTOGRAM;
+            case OBSERVABLE_GAUGE:
+              return LAST_VALUE;
+          }
+          logger.log(
+              Level.WARNING, "Unable to find default aggregation for instrument: " + instrument);
+          return NONE;
+        }
+
+        @Override
+        public <T> Aggregator<T> createAggregator(
+            Resource resource,
+            InstrumentationLibraryInfo instrumentationLibraryInfo,
+            InstrumentDescriptor instrumentDescriptor,
+            MetricDescriptor metricDescriptor,
+            ExemplarFilter exemplarFilter) {
+          return resolve(instrumentDescriptor)
+              .createAggregator(
+                  resource,
+                  instrumentationLibraryInfo,
+                  instrumentDescriptor,
+                  metricDescriptor,
+                  exemplarFilter);
+        }
+
+        @Override
+        public String toString() {
+          return "default";
+        }
+      };
   private static final Aggregation SUM = sum(AggregationTemporality.CUMULATIVE);
-  private static final Aggregation LAST_VALUE =
-      Aggregation.make("lastValue", unused -> AggregatorFactory.lastValue());
+  private static final Aggregation LAST_VALUE = new LastValueAggregation();
```

Comment:
same warning here about classloading deadlock

Reply:
if you make these built-ins be private static inner classes of this class, the warning clears up. Just an idea.

Reply:
I moved all static class initialization to only use "upstream" dependencies instead of circular like it is here.